eth_tester/main.py

```diff
@@ -333,9 +333,13 @@ def get_transaction_receipt(self, transaction_hash):
         raw_transaction_hash = self.normalizer.normalize_inbound_transaction_hash(
             transaction_hash,
         )
+        transaction = self.get_transaction_by_hash(transaction_hash)
         raw_receipt = self.backend.get_transaction_receipt(raw_transaction_hash)
         self.validator.validate_outbound_receipt(raw_receipt)
         receipt = self.normalizer.normalize_outbound_receipt(raw_receipt)
+        byzantium_fork_block = self.get_fork_block('FORK_BYZANTIUM')
+        if byzantium_fork_block is None or byzantium_fork_block > transaction['block_number']:
+            del receipt['status']
```

Comment:
Instead of removing the `status` field here, what about changing this so that it instead checks to see if we are in the byzantium vm and if `True`, removes the `state_root` field and adds the `status` field.  That would remove any need to change implementation in individual backends.

Reply:
the main problem I've run into are validators - there is no easy way of having them check the status conditionally and the status must be set by the serializers of individual backends.

Reply:
So my thought is that:

1. All of the backends would **not** set status.
2. Any validators/normalizers that apply at the backend level get stripped of any knowledge of the `status` field
3. All backends include `state_root` (which for byzantium and later will always bee `0x0` or `0x1`.
4. At the `EthereumTester` level, *if* we are in the byzantium rules, we pop the `state_root` off of the return value, and inject a new `status` field computed from the `state_root` field.

Reply:
Please check the diff now, I've updated it according to your comments.